maestro-client/src/main/java/maestro/drivers/IOSDriver.kt

```diff
@@ -325,65 +315,104 @@ class IOSDriver(
+                start.distance(end) / toSeconds(durationMs)
+            } else {
+                Float.MAX_VALUE
+            }
         ).expect {}
     }
 
     override fun swipe(swipeDirection: SwipeDirection, durationMs: Long) {
-        val width = widthPixels ?: throw IllegalStateException("Device width not available")
-        val height = heightPixels ?: throw IllegalStateException("Device height not available")
+        val width = widthPoints ?: throw IllegalStateException("Device width not available")
+        val height = heightPoints ?: throw IllegalStateException("Device height not available")
+
+        val startPoint: PointF
+        val endPoint: PointF
 
         when (swipeDirection) {
             SwipeDirection.UP -> {
-                iosDevice.scroll(
-                    xStart = width / 2,
-                    yStart = height / 3,
-                    xEnd = width / 2,
-                    yEnd = height / 9,
-                    durationMs = durationMs,
-                    IdbIOSDevice.ScrollType.SWIPE
-                ).expect {}
+                startPoint = PointF(
+                    x = 0.5f,
+                    y = 0.5f,
+                )
+                endPoint = PointF(
+                    x = 0.5F,
+                    y = 0.25f,
+                )
             }
             SwipeDirection.DOWN -> {
-                iosDevice.scroll(
-                    xStart = width / 2,
-                    yStart = height / 9,
-                    xEnd = width / 2,
-                    yEnd = height / 3,
-                    durationMs = durationMs,
-                    IdbIOSDevice.ScrollType.SWIPE
-                ).expect {}
+                startPoint = PointF(
+                    x = 0.5f,
+                    y = 0.5f,
+                )
+                endPoint = PointF(
+                    x = 0.5F,
+                    y = 0.75f,
+                )
             }
             SwipeDirection.RIGHT -> {
-                iosDevice.scroll(
-                    xStart = 0,
-                    yStart = height / 4,
-                    xEnd = width / 4,
-                    yEnd = height / 4,
-                    durationMs = durationMs,
-                    IdbIOSDevice.ScrollType.SWIPE
-                ).expect {}
+                startPoint = PointF(
+                    x = 0.5f,
+                    y = 0.5f,
+                )
+                endPoint = PointF(
+                    x = 0.9F,
+                    y = 0.5f,
+                )
             }
             SwipeDirection.LEFT -> {
-                iosDevice.scroll(
-                    xStart = width / 4,
-                    yStart = height / 4,
-                    xEnd = 0,
-                    yEnd = height / 4,
-                    durationMs = durationMs,
-                    IdbIOSDevice.ScrollType.SWIPE
-                ).expect {}
+                startPoint = PointF(
+                    x = 0.5f,
```

Comment:
I think we might need to directional swipe from 1 to 0 or 1 to 0.1 and vice versa for left and right. Following is the result with the current checked on two simulators:

https://user-images.githubusercontent.com/12881364/211326044-c77b2fda-8db0-42f1-9fc1-8f08d2b7d1f0.mp4

Reply:
Should we also do it for swipe up and down?

Reply:
Good call 👍

Reply:
Let's make sure it is updated for Android too

Reply:
I'll do it in my PR

Reply:
https://github.com/mobile-dev-inc/maestro/pull/585